spectrumx_visualization_platform/spx_vis/api/views.py

```diff
@@ -1,119 +1,212 @@
 from django.http import FileResponse
-from rest_framework import filters
-from rest_framework import permissions
-from rest_framework import status
-from rest_framework import viewsets
-from rest_framework.decorators import action
-from rest_framework.parsers import FormParser
-from rest_framework.parsers import MultiPartParser
+from rest_framework import filters, permissions, status, viewsets
+from rest_framework.decorators import action, api_view
+from rest_framework.parsers import FormParser, MultiPartParser
+from rest_framework.request import Request
 from rest_framework.response import Response
-
-from spectrumx_visualization_platform.spx_vis.api.serializers import CaptureSerializer
-from spectrumx_visualization_platform.spx_vis.api.serializers import FileSerializer
+from spectrumx.errors import FileError
+from spectrumx.models.captures import CaptureType
+from spectrumx_visualization_platform.spx_vis.api.serializers import CaptureSerializer, FileSerializer
 from spectrumx_visualization_platform.spx_vis.capture_utils.sigmf import SigMFUtility
-from spectrumx_visualization_platform.spx_vis.models import Capture
-from spectrumx_visualization_platform.spx_vis.models import File
+from spectrumx_visualization_platform.spx_vis.models import Capture, File
+
+from datetime import datetime, timedelta
+
+@api_view(["GET"])
+def capture_list(request: Request) -> Response:
+    """Get the list of captures for the current user."""
+    file_list = []
+    sds_file_list = []
+    sds_file_count = 0
+
+    print("i was here")
+
+    sds_client = request.user.sds_client()
+    if sds_client:
+        print("Successfully connected to SDS client")
+    else:
+        print("Connection error")
+
+    # Get the list of files from the SDS client
+    try:
+        sds_captures = sds_client.captures.listing(capture_type=CaptureType.RadioHound)
+        sds_file_count = len(sds_captures)
+        print(f"Number of captures from SDS client: {sds_file_count}")
+    except FileError:
+        sds_captures = []
+        print("No captures found in SDS Client.")
+
+    # Process SDS captures
+    for sds_capture in sds_captures:
+        timestamp = sds_capture.get("timestamp", "")
+        scan_time = sds_capture.get("metadata", {}).get("scan_time", 0.08121538162231445)  # Default value for test
+        end_time = calculate_end_time(timestamp, scan_time)  # Compute end time
+
+        print(f"SDS Capture Debug: timestamp={timestamp}, scan_time={scan_time}, computed end_time={end_time}")
+
+        sds_file_list.append({
+            "id": sds_capture.get("_id", sds_capture.get("id", "unknown")),
+            "name": sds_capture["name"],
+            "media_type": sds_capture.get("metadata", {}).get("data_type", "unknown"),
+            "timestamp": timestamp,
+            "created_at": capture.get("timestamp", ""),
```

Comment:
Does the SDS have separate fields for timestamp and creation date? Because we think of those as different things--the timestamp is when the data was recorded by a radio device, whereas the timestamp is when the SDS capture object was created. Not sure if the SDS uses the same definitions. Something to ask Lucas

Reply:
Don't worry about this for now though